### PR TITLE
T34363 final hook

### PR DESCRIPTION
**Pull request by manuq**
https://phabricator.endlessm.com/T34363

### CODE REVIEW

**Comment by manuq** [comment]
Note that this is on top of https://github.com/endlessm/kolibri-explore-plugin/pull/496 which at the same time is on top of https://github.com/endlessm/kolibri-explore-plugin/pull/495

**Review by dylanmccall — APPROVED** [review]
A couple comments, but all's good :+1:

File: kolibri_explore_plugin/collectionviews.py [hunk 1]
```diff
@@ -518,16 +512,19 @@ def _read_content_manifests():
...
+            logger.error(err)
+        else:
+            manifest.set_availability(free_space_gb)
+            _content_manifests.append(manifest)
 
-    # FIXME: Adding the sample starter pack in a hacky way for now.
-    _create_manifest("explorer", "0001")
+            if grade not in _content_manifests_by_grade_name:
+                _content_manifests_by_grade_name[grade] = {}
+            _content_manifests_by_grade_name[grade][name] = manifest
```
[the remark this answers is not in the document]
Comment (manuq) [reply kolibri_explore_plugin/collectionviews.py:527]: Oh good point! I will use `setdefault` to remove the conditional and leave the second line so it's more readable vs the one-liner:
```
            _content_manifests_by_grade_name.setdefault(grade, {})
            _content_manifests_by_grade_name[grade][name] = manifest
```

File: kolibri_explore_plugin/collectionviews.py [hunk 2]
```diff
@@ -555,9 +552,12 @@ def get_collections_info(request):
...
-            "metadata": GRADES_METADATA[grade],
         }
         for name in COLLECTION_NAMES:
+            if grade not in _content_manifests_by_grade_name:
+                continue
+            if name not in _content_manifests_by_grade_name[grade]:
+                continue
             manifest = _content_manifests_by_grade_name[grade][name]
             collection_info = {
                 "grade": manifest.grade,
```
[the remark this answers is not in the document]
Comment (manuq) [reply kolibri_explore_plugin/collectionviews.py:563]: Good point! I will do so. The only reason I added these for/continue is to make the code more robust to collection changes. Eg. if the collections repo doesn't have an "scientist-0001.json" file the append was crashing when looping with grade="scientist" and name="0001". Now I added all collection files so this shouldn't happen but I think is good to keep the check.

**Pull request merged**